src/aws-cpp-sdk-core/source/smithy/client/AwsSmithyClientBase.cpp

```diff
@@ -314,20 +314,20 @@ void AwsSmithyClientBase::HandleAsyncReply(std::shared_ptr<AwsSmithyClientAsyncR
         }
     };
 
-    bool hasEmbeddedError = pRequestCtx->m_pRequest &&
-                          pRequestCtx->m_pRequest->HasEmbeddedError(httpResponse->GetResponseBody(), httpResponse->GetHeaders());
-
-    if (Utils::DoesResponseGenerateError(httpResponse) || hasEmbeddedError)
+    Aws::Client::HttpResponseOutcome outcome = [&]()
     {
-        AWS_LOGSTREAM_DEBUG(AWS_SMITHY_CLIENT_LOG, "Request returned error. Attempting to generate appropriate error codes from response");
-        assert(m_errorMarshaller);
-        auto error = m_errorMarshaller->BuildAWSError(httpResponse);
-        return pRequestCtx->m_responseHandler(HttpResponseOutcome(std::move(error)));
-    }
-
-    AWS_LOGSTREAM_DEBUG(AWS_SMITHY_CLIENT_LOG, "Request returned successful response.");
-
-    Aws::Client::HttpResponseOutcome outcome = HttpResponseOutcome(std::move(httpResponse));
+        bool hasEmbeddedError = pRequestCtx->m_pRequest &&
+                      pRequestCtx->m_pRequest->HasEmbeddedError(httpResponse->GetResponseBody(), httpResponse->GetHeaders());
+        if (Utils::DoesResponseGenerateError(httpResponse) || hasEmbeddedError)
+        {
+            AWS_LOGSTREAM_DEBUG(AWS_SMITHY_CLIENT_LOG, "Request returned error. Attempting to generate appropriate error codes from response");
+            assert(m_errorMarshaller);
+            auto error = m_errorMarshaller->BuildAWSError(httpResponse);
+            return HttpResponseOutcome(std::move(error));
```

Comment:
the main diff here is to not perform early return 
```
return pRequestCtx->m_responseHandler(HttpResponseOutcome(std::move(error)));
```

but I also used IIFE to avoid modification of the other blocks by keeping the original variable `Aws::Client::HttpResponseOutcome outcome`